### PR TITLE
[CFM-1434] Redirect node/$group_id to dashboard.

### PR DESCRIPTION
**Pull request by ybaras**
https://issuetracker.amplexor.com/jira/browse/CFM-1434

### CODE REVIEW

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_group/c4m_content_group.module [hunk 1]
```diff
@@ -22,6 +22,49 @@ define('C4M_CONTENT_GROUP_STATUS_UPDATE_DELETED', 11);
...
+    return;
+  }
+
+  $node = $item['map'][1];
+
+  if (!og_is_group_type('node', $node->type)) {
+    return;
+  }
+
+  $path = $node->purl;
```
Comment (amitaibu) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_group/c4m_content_group.module]: Add comment

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_group/c4m_content_group.module [hunk 2]
```diff
@@ -22,6 +22,49 @@ define('C4M_CONTENT_GROUP_STATUS_UPDATE_DELETED', 11);
...
+    return;
+  }
+
+  $node = $item['map'][1];
+
+  if (!og_is_group_type('node', $node->type)) {
+    return;
+  }
+
+  drupal_goto($node->purl);
```
Comment (amitaibu) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_group/c4m_content_group.module]: this is likely to cause an endless loop

Comment (amitaibu) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_group/c4m_content_group.module]: also these hooks probably need to be in `c4m_og` module

**Comment by amitaibu** [comment]
Looks good, but missing a screenshot to show it to everyone; )

File: project/profiles/capacity4more/modules/c4m/og/c4m_og_purl/c4m_og_purl.module [hunk 3]
```diff
@@ -13,6 +13,60 @@ include_once 'c4m_og_purl.ds_render_field.inc';
 include_once 'c4m_og_purl.uri_info.inc';
 
 /**
+ * Implements hook_init().
+ *
+ * Redirect /node/GROUP_ID path to the group dashboard /GROUP-TITLE.
+ */
+function c4m_og_purl_init() {
+  $item = menu_get_item();
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og_purl/c4m_og_purl.module]: @amitaibu 
Isn't it better to use `menu_get_object()` in here ?

File: project/profiles/capacity4more/modules/c4m/og/c4m_og_purl/c4m_og_purl.module [hunk 4]
```diff
@@ -13,6 +13,60 @@ include_once 'c4m_og_purl.ds_render_field.inc';
...
+  $node = $item['map'][1];
+
+  if (!og_is_group_type('node', $node->type)) {
+    // Node is not a group.
+    return;
+  }
+
+  // Reset the path to point to the root.
+  $path = '';
+  $options['alias'] = $path;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og_purl/c4m_og_purl.module]: @ybaras 
Can't we just use one line over here, like:
```PHP
$options['alias'] = '';
```
?

**Comment by ybaras** [comment]
Screenshot of a request to a group node id `/node/13` which gets redirected to the group dashboard at `/music-lovers`.

test-record

**Comment by ybaras** [comment]
Path alias (for `url()`) is also set to the group dashboard:

screenshot from 2016-06-26 10 08 02

File: project/profiles/capacity4more/modules/c4m/og/c4m_og_purl/c4m_og_purl.module [hunk 5]
```diff
@@ -18,21 +18,24 @@ include_once 'c4m_og_purl.uri_info.inc';
  * Redirect /node/GROUP_ID path to the group dashboard /GROUP-TITLE.
  */
 function c4m_og_purl_init() {
-  $item = menu_get_item();
+  $item = menu_get_object();
```
Comment (amitaibu) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og_purl/c4m_og_purl.module]: This should be called $node

**Comment by ybaras** [comment]
@ordavidil `menu_get_object() -> book['router_path']` also redirects `/node/%/edit`.
This is why Behat currently fails.
I am switching back to `menu_get_item()['path']`.

File: project/profiles/capacity4more/modules/c4m/og/c4m_og_purl/c4m_og_purl.module [hunk 6]
```diff
@@ -13,6 +13,61 @@ include_once 'c4m_og_purl.ds_render_field.inc';
...
+  $node = $item['map'][1];
+
+  if (!og_is_group_type('node', $node->type)) {
+    // This is not a group.
+    return;
+  }
+
+  // Set the path alias.
+  $path = $node->purl;
+  $options['alias'] = $path;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og_purl/c4m_og_purl.module]: @ybaras 
Please change this to `$options['alias'] = $node->purl;`

**Pull request merged**